src/axom/quest/detail/MeshTester_detail.hpp

```diff
@@ -0,0 +1,447 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_QUEST_MESH_TESTER_DETAIL_HPP_
+#define AXOM_QUEST_MESH_TESTER_DETAIL_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/primal.hpp"
+#include "axom/spin.hpp"
+#include "axom/mint.hpp"
+
+// RAJA includes
+#if defined(AXOM_USE_RAJA)
+  #include "RAJA/RAJA.hpp"
+  #include "axom/mint/execution/internal/structured_exec.hpp"
+#endif
+
+// Acceleration data structure includes
+#include "axom/spin/BVH.hpp"
+#include "axom/spin/ImplicitGrid.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace detail
+{
+using UMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
+using Triangle3 = primal::Triangle<double, 3>;
+using SpatialBoundingBox = primal::BoundingBox<double, 3>;
+using UniformGrid3 = spin::UniformGrid<int, 3>;
+using Point3 = primal::Point<double, 3>;
+
+inline detail::Triangle3 getMeshTriangle(axom::IndexType i,
+                                         detail::UMesh* surface_mesh)
+{
+  SLIC_ASSERT(surface_mesh->getNumberOfCellNodes(i) == 3);
+
+  detail::Triangle3 tri;
+
+  const axom::IndexType* triCell = surface_mesh->getCellNodeIDs(i);
+
+  const double* x = surface_mesh->getCoordinateArray(mint::X_COORDINATE);
+  const double* y = surface_mesh->getCoordinateArray(mint::Y_COORDINATE);
+  const double* z = surface_mesh->getCoordinateArray(mint::Z_COORDINATE);
+
+  for(int n = 0; n < 3; ++n)
+  {
+    const axom::IndexType nodeIdx = triCell[n];
+    tri[n][0] = x[nodeIdx];
+    tri[n][1] = y[nodeIdx];
+    tri[n][2] = z[nodeIdx];
+  }
```

Comment:
Minor: This might be a bit cleaner as:
```suggestion
  for(int n = 0; n < 3; ++n)
  {
    const axom::IndexType nodeIdx = triCell[n];
    tri[n] = Point3{ x[nodeIdx], y[nodeIdx], z[nodeIdx]};
  }
```